src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -28,6 +29,94 @@ static ucs_config_field_t uct_cuda_ipc_md_config_table[] = {
     {NULL}
 };
 
+typedef struct {
+    /* GPU Device number */
+    int     dev_num;
+    /* Cache of accessible devices (ucs_ternary_auto_value_t) */
+    uint8_t accessible[0];
+} uct_cuda_ipc_dev_cache_t;
+
+static UCS_F_ALWAYS_INLINE int uct_cuda_ipc_uuid_equals(CUuuid a, CUuuid b)
+{
+    int64_t *a64 = (int64_t *)a.bytes;
+    int64_t *b64 = (int64_t *)b.bytes;
+    return (a64[0] == b64[0]) && (a64[1] == b64[1]);
+}
+
+static UCS_F_ALWAYS_INLINE khint32_t uct_cuda_ipc_uuid_hash_func(CUuuid key)
+{
+    int64_t *i64 = (int64_t *)key.bytes;
+    return (khint32_t)(i64[0]);
+}
+
+KHASH_INIT(cuda_ipc_uuid_hash, CUuuid, uct_cuda_ipc_dev_cache_t*, 1,
+           uct_cuda_ipc_uuid_hash_func, uct_cuda_ipc_uuid_equals);
+
+typedef struct {
+    khash_t(cuda_ipc_uuid_hash) hash;
+    ucs_recursive_spinlock_t    lock;
+} uct_cuda_ipc_uuid_cache_t;
+
+static uct_cuda_ipc_uuid_cache_t uct_cuda_ipc_uuid_cache;
+
+static uct_cuda_ipc_dev_cache_t *uct_cuda_ipc_create_dev_cache(int dev_num)
+{
+    uct_cuda_ipc_dev_cache_t *cache;
+    ucs_status_t status;
+    int i, num_devices;
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuDeviceGetCount(&num_devices));
```

Comment:
The only reason is that `UCT_CUDA_IPC_DEVICE_GET_COUNT` returns ucs_status_t in case of an error, while this function returns a pointer to the created cache.
I could use this macro only if convert function to return status, in UCX code we do it both ways, so I preferred the one with shorter signature. But it you prefer to return a status - please let me know